File: src/j/implementation/microSet/InnerSetIterator.java
package j.implementation.microSet;

import java.util.Iterator;

public class InnerSetIterator<T> implements Iterator<T>
{
	private int index = 0;
	private InnerSet innerSet;
	
	public InnerSetIterator(InnerSet<T> p_innerSet)
	{
		innerSet = p_innerSet;
	}

	@Override
	public boolean hasNext()
	{
		return index != innerSet.getSize();
	}

	@Override
	public T next()
	{
		return (T) innerSet.getElement(index++);
	}
}

File: src/scala/implementation/InnerSetIterator.scala
package scala.implementation

import java.util.Iterator

/**
  * Created by pxl on 29/03/17.
  */


class InnerSetIterator[T](innerSet: InnerSet[T]) extends Iterator[T] {

  var index: Int = 0

  override def next(): T =
  {
    index += 1
    val t = innerSet.getElement(index)
    t.get
  }

  override def hasNext: Boolean = index != innerSet.getSize
}
